pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -231,6 +247,11 @@ func (l *podNIC) PlugPhase1() error {
 			}
 		}
 
+		if err := bindMechanism.preparePodNetworkInterface(); err != nil {
```

Comment:
Since the `bindMechanism.generateDomainIfaceSpec()` it makes reading this code very confusing.
Maybe we can generate both the `domain` and `dhcp` configs after the discover, but doing the `Export` part after the prepare.

Anyway, I would add a comment explaining that the config should be generated before calling to `prepare`.

Reply:
I disagree. 

What makes this code confusing, hard to reason about, and totally dependent on the order of operations is the fact that we replace the pod nic with a dummy.

I'll add the comment.

Reply:
What I tried to say is that it is very hard for the reader of the code to understand the order of the `generate`, `prepare`, `export` and the fact the order is important.

For the `dhcpConfig` we `generate` + `export` before the `prepare`. While for the `domainInterface` we do it after the `preapare`.
What I'm suggesting is -
Option 1 - `generate` both `dhcpConfig` and `domainInterface` before the `prepare` and storing both after the `prepare`.
So it will be clear the order is `dicover` -> `generate` -> `prepare` -> `export`.
Option 2 - Add comments  explaining the current order.
2.

Reply:
Added a comment.

Let me know if it explains it correctly.